mlflow/R/mlflow/R/tracking-runs.R

```diff
@@ -0,0 +1,597 @@
+#' attached to it, for instance:
+#'
+#' \code{
+#' {
+#' "Version": "2012-10-17",
+#' "Statement": [
+#'   {
+#'     "Sid": "VisualEditor0",
+#'     "Effect": "Allow",
+#'     "Action": [
+#'       "s3:PutObject",
+#'       "s3:GetObject",
+#'       "s3:ListBucket",
+#'       "s3:GetBucketLocation"
+#'       ],
+#'     "Resource": [
+#'       "arn:aws:s3:::mlflow-test/*",
+#'       "arn:aws:s3:::mlflow-test"
+#'       ]
+#'   }
+#'   ]
+#' }
+#' }
+#'
+#' Additionally, at least the \code{AWS_ACCESS_KEY_ID} and \code{AWS_SECRET_ACCESS_KEY}
+#' environment variables must be set to the corresponding key and secrets provided
+#' by Amazon IAM.
+#'
+#' @export
+mlflow_log_artifact <- function(path, artifact_path = NULL, run_id = NULL, client = NULL) {
+  c(client, run_id) %<-% resolve_client_and_run_id(client, run_id)
+  artifact_param <- NULL
+  if (!is.null(artifact_path)) artifact_param <- "--artifact-path"
+
+  if (as.logical(fs::is_file(path))) {
+    command <- "log-artifact"
+    local_param <- "--local-file"
+  } else {
+    command <- "log-artifacts"
+    local_param <- "--local-dir"
+  }
+
+  mlflow_cli("artifacts",
+             command,
+             local_param,
+             path,
+             artifact_param,
+             artifact_path,
+             "--run-id",
+             run_id,
+             client = client
+  )
+
+  mlflow_list_artifacts(run_id = run_id, path = artifact_path, client = client)
+}
+
+
+#' Start Run
+#'
+#' Starts a new run. If `client` is not provided, this function also registers the created run as the active run.
+#'
+#' @param run_uuid If specified, get the run with the specified UUID and log metrics
+#'   and params under that run. The run's end time is unset and its status is set to
+#'   running, but the run's other attributes remain unchanged.
+#' @param experiment_id Used only when `run_uuid` is unspecified. ID of the experiment under
+#'   which to create the current run. If unspecified, the run is created under
+#'   a new experiment with a randomly generated name.
+#' @param source_name Name of the source file or URI of the project to be associated with the run.
+#'   Defaults to the current file if none provided.
+#' @param source_version Optional Git commit hash to associate with the run.
+#' @param entry_point_name Optional name of the entry point for to the current run.
+#' @param source_type Integer enum value describing the type of the run  ("local", "project", etc.).
+#' @param user_id User ID or LDAP for the user executing the run. Only used when `client` is specified.
+#' @param run_name Human readable name for run. Only used when `client` is specified.
+#' @param start_time Unix timestamp of when the run started in milliseconds. Only used when `client` is specified.
+#' @param tags Additional metadata for run in key-value pairs. Only used when `client` is specified.
+#' @template roxlate-client
+#'
+#' @examples
+#' \dontrun{
+#' with(mlflow_start_run(), {
```

Comment:
should this be with.mlflow_run?

Reply:
`mlflow_start_run()` returns an `mlflow_run` object so `with` will dispatch accordingly